File: python/2696. Minimum String Length After Removing Substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        for ch in s:
            if stack and (
                (ch == "B" and stack[-1] == "A") or (ch == "D" and stack[-1] == "C")
            ):
                stack.pop()
            else:
                stack.append(ch)
        return len(stack)

File: rust/2696. Minimum String Length After Removing Substrings.rs
impl Solution {
    pub fn min_length(s: String) -> i32 {
        let mut vec = Vec::new();
        for ch in s.chars() {
            if !vec.is_empty()
                && ((ch == 'B' && *vec.last().unwrap() == 'A')
                    || (ch == 'D' && *vec.last().unwrap() == 'C'))
            {
                vec.pop();
            } else {
                vec.push(ch);
            }
        }
        vec.len() as i32
    }
}
